api/src/main/java/org/pmiops/workbench/google/CloudStorageServiceImpl.java

```diff
@@ -25,10 +30,26 @@ public String readBlockscoreApiKey() {
     return readToString(getCredentialsBucketName(), "blockscore-api-key.txt").trim();
   }
 
+  @Override
+  public List<FileDetail> getBucketFileList(String bucketName) {
+    List<FileDetail> fileList = new ArrayList<FileDetail>();
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+    Iterable<Blob> blobList =storage.get(bucketName).list().getValues();
+    blobList.forEach(blobItem->{
+      FileDetail fileDetail = new FileDetail();
+      fileDetail.setName(blobItem.getName());
+      fileDetail.setUrl(blobItem.getMediaLink());
```

Comment:
Do you have an example of this value? I'm thinking media link may not be what we'd want, but I'm not sure. Most likely we want a `gs://` prefixed URL to interface with the notebook service, I'd guess.

Reply:
Right so while saving the file back to notebook server the pattern needed is gs://<firecloudbucket>/fileName. I am going to change the logic here to have that as the path so that we do not need extra logic later